contrib/node/src/python/pants/contrib/node/subsystems/eslint.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import filecmp
+import logging
+import os.path
+import shutil
+from typing import Tuple
+
+from pants.option.custom_types import dir_option, file_option
+from pants.subsystem.subsystem import Subsystem
+from pants.util.dirutil import safe_mkdir, safe_rmtree
+
+
+logger = logging.getLogger(__name__)
+
+
+class ESLint(Subsystem):
+  options_scope = 'eslint'
+
+  required_files = ['yarn.lock', 'package.json']
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register('--version',  default='4.15.0', fingerprint=True, help='Use this ESLint version.')
+    register(
+      '--config', type=file_option, default=None,
+      help="Path to `.eslintrc` or alternative ESLint config file",
+    )
+    register(
+      '--setupdir', type=dir_option, fingerprint=True,
+      help='Find the package.json and yarn.lock under this dir for installing eslint and plugins.',
+    )
+    register(
+      '--ignore', type=file_option, fingerprint=True,
+      help='The path to the global eslint ignore path',
+    )
+
+  def configure(self, *, bootstrapped_support_path: str) -> None:
+    logger.debug(
+      f'Copying {self.options.setupdir} to bootstrapped dir: {bootstrapped_support_path}'
+    )
+    safe_rmtree(bootstrapped_support_path)
+    shutil.copytree(self.options.setupdir, bootstrapped_support_path)
+
+  def supportdir(self, *, task_workdir: str) -> Tuple[str, bool]:
+
```

Comment:
Stray newline.

If you end up giving this another run through CI, you might also suggest adding a warning that this `supportdir` method and strategy should only be used under v1.

Reply:
Good catch. Will do in a followup PR.